packages/sandcastle/src/Helpers.ts

```diff
@@ -0,0 +1,81 @@
+import Pako from "pako";
+
+export function embedInSandcastleTemplate(code: string, addExtraLine: boolean) {
+  console.log("embedSandcastle");
+  return `window.startup = async function (Cesium, Sandcastle) {
+  'use strict';
+  //Sandcastle_Begin
+  ${addExtraLine ? "\n" : ""}${code}
+  //Sandcastle_End
+  Sandcastle.finishedLoading();
+  };
+  if (typeof Cesium !== 'undefined' && typeof Sandcastle !== 'undefined') {
+      window.startupCalled = true;
+    window.startup(Cesium, Sandcastle).catch((error) => {
+        "use strict";
+      console.error(error);
+    });
+}
+`;
+}
+
+type SandcastleSaveData = {
+  code: string;
+  html: string;
+  baseHref?: string;
+};
+
+export function makeCompressedBase64String(data: SandcastleSaveData) {
+  // data stored in the hash as:
+  // Base64 encoded, raw DEFLATE compressed JSON array where index 0 is code, index 1 is html
+  const { code, html, baseHref } = data;
+  const encode = baseHref !== undefined ? [code, html, baseHref] : [code, html];
```

Comment:
```suggestion
  const encode = baseHref ? [code, html, baseHref] : [code, html];
```

Since this is now typed, be don't need to worry about `null`.